pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -186,22 +187,22 @@ func ensureTeamUsersAreInSync(workflowCtx *workflow.Context, teamID string, team
 		usernamesMap[string(username)] = struct{}{}
 	}
 
-	atlasUsernamesMap := map[string]mongodbatlas.AtlasUser{}
-	for _, atlasUser := range atlasUsers {
+	atlasUsernamesMap := map[string]admin.CloudAppUser{}
+	for _, atlasUser := range atlasUsers.GetResults() {
 		atlasUsernamesMap[atlasUser.Username] = atlasUser
 	}
 
 	g, taskContext := errgroup.WithContext(workflowCtx.Context)
 
-	for i := range atlasUsers {
-		user := atlasUsers[i]
-		if _, ok := usernamesMap[atlasUsers[i].Username]; !ok {
-			g.Go(func() error {
-				workflowCtx.Log.Debugf("removing user %s from team %s", user.ID, teamID)
-				_, err := workflowCtx.Client.Teams.RemoveUserToTeam(taskContext, workflowCtx.OrgID, teamID, user.ID)
-
-				return err
-			})
+	if atlasUsers.Results != nil {
+		for _, user := range atlasUsers.GetResults() {
+			if _, ok := usernamesMap[user.Username]; !ok {
+				g.Go(func() error {
```

Comment:
Not sure why we added concurrency here, we are launching a series of deletions that we are going to wait for. Are they really so slow? Only if this takes longer than 5-15 seconds total it might make sense.

Anyway, this is not a comment about this PR in particular.

Reply:
I'm not sure I understand what you mean by this, could you expand? I'm not sure how this constitutes as concurrency.

Reply:
`g.Go(func() error {`
https://pkg.go.dev/golang.org/x/sync/errgroup#Group.Go
This launches a **goroutine** on an concurrent `errgroup` with the main thread. Then the main thread will check all the returns to see if there are any errors from any of the goroutines.

I wonder how useful or required it is, the only gain is that you wait only for the slowest request, but this does not seem critical here, unless I am missing something.

Again not a critique to this PR. Just noticed we were doing this, I was not aware before and seems odd.

Reply:
I see. The implementation of the translation layer would be a natural place to reconsider this approach. I'll bring this up in one of the upcoming sync-up optionals to see if the rest of the team has any insights on this detail.

Reply:
The request is usually fast, but a team can be a single user or an entire org. As there's no bulk deletion on the API this is just to be a little quicker in the worst scenario. I don't get why it is odd.

Reply:
Rule of thumb is avoid concurrency unless absolutely necessary, it adds complexity and possible race conditions hard to debug, etc. I do not see a particular risk here, we are just removing and reading back the results, but still I do not see the gain either, measured against the added complexity. That is all.